tools/semver/semver.go

```diff
@@ -0,0 +1,89 @@
+// Package semver implements comparison of semantic version strings.
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/mod/semver"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		usage()
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case "check":
+		if len(os.Args) != 3 {
+			fmt.Fprintf(os.Stderr, "Error: 'check' requires exactly one version argument\n")
+			usage()
+			os.Exit(1)
+		}
+		checkVersion(os.Args[2])
+
+	case "compare":
+		if len(os.Args) != 4 {
+			fmt.Fprintf(os.Stderr, "Error: 'compare' requires exactly two version arguments\n")
+			usage()
+			os.Exit(1)
+		}
+		compareVersions(os.Args[2], os.Args[3])
+
+	default:
+		fmt.Fprintf(os.Stderr, "Error: unknown command %q\n", os.Args[1])
+		usage()
+		os.Exit(1)
+	}
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, `Usage: %s <command> [arguments]
+
+Commands:
+  check <version>           Check if a version is valid
+  compare <ver1> <ver2>     Compare two versions
+
+Examples:
+  %s check 1.2.3           # Prints "valid" or "invalid"
+  %s compare 1.2.3 2.0.0   # Prints "less", "equal", or "greater"
+`, os.Args[0], os.Args[0], os.Args[0])
```

Comment:
I think you can replace the stuttering of `os.Args[0]` with `%[0]s` for each elements to replace

Reply:
done